src/mappings/mappingHandlers.ts

```diff
@@ -49,6 +66,98 @@ BigInt.prototype.toJSON = function () {
   return this.toString();
 };
 
+async function saveIbcChannel(channelName: string) {
+  const generatedEscrowAddress = getEscrowAddress(TRANSFER_PORT_VALUE, channelName);
+
+  const channelRecord = new IBCChannel(channelName, channelName, generatedEscrowAddress);
+  channelRecord.save();
+}
+
+export async function handleIbcSendPacketEvent(cosmosEvent: CosmosEvent): Promise<void> {
+  const { event, block, tx } = cosmosEvent;
+  if (event.type != EVENT_TYPES.SEND_PACKET) {
+    logger.warn("Not valid send_packet event.");
+    return;
+  }
+
+  const packetSrcPortAttr = event.attributes.find((a) => a.key === PACKET_SRC_PORT_KEY);
+  if (!packetSrcPortAttr || packetSrcPortAttr.value !== TRANSFER_PORT_VALUE) {
+    logger.warn("packet_src_port is not transfer");
+    return;
+  }
+  const packetDataAttr = event.attributes.find((a) => a.key === PACKET_DATA_KEY);
+  if (!packetDataAttr) {
+    logger.warn("No packet data attribute found");
+    return;
+  }
+
+  const packetSrcChannelAttr = event.attributes.find((a) => a.key === PACKET_SRC_CHANNEL_KEY);
+  if (!packetSrcChannelAttr) {
+    logger.warn("No packet source channel found");
+    return;
+  }
+  const { amount, denom, receiver, sender } = JSON.parse(packetDataAttr.value);
+  const sourceChannel = packetSrcChannelAttr.value;
+
+  saveIbcChannel(sourceChannel);
+
+  const transferRecord = new IBCTransfer(
+    tx.hash,
+    block.header.time as any,
+    BigInt(block.header.height),
+    packetSrcChannelAttr.value,
+    sender,
+    receiver,
+    denom,
+    amount,
+    TransferType.SEND,
+  );
+  transferRecord.save();
```

Comment:
Should await on the returned value of `.save()` otherwise the record might not get saved.